File: cfgs/java_cfgs_foo/COUNT_DERANGEMENTS_PERMUTATION_SUCH_THAT_NO_ELEMENT_APPEARS_IN_ITS_ORIGINAL_POSITION_1.java
Nodes
1;special;;;
2;statement;int[] der = new int[foo_1 + 1];14;
3;statement;foo_1[0] = 1;15;
4;statement;foo_1[1] = 0;16;
5;statement;foo_1[2] = 1;17;
6;statement;int i = 3;18;
7;conditional;foo_1 <= foo_2;19;
8;statement;foo_1[foo_2] = (foo_3 - 1) * (foo_4[foo_5 - 1] + foo_6[foo_7 - 2]);20;
9;statement;++i;20;
10;statement;return foo_1[foo_2];21;
11;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;
5;7;
6;8;10;
7;9;
8;7;
9;11;
10;

File: cfgs/pythoncfgs/COUNT_DERANGEMENTS_PERMUTATION_SUCH_THAT_NO_ELEMENT_APPEARS_IN_ITS_ORIGINAL_POSITION_1.py
Nodes
1;statement;foo_1 = [0 for foo_2 in fun_1(foo_3 + 1)];8;
2;statement;foo_1[0] = 1;9;
3;statement;foo_1[1] = 0;10;
4;statement;foo_1[2] = 1;11;
5;conditional;foo < foo;12;
6;statement;return foo_1[foo_2];14;
7;statement;foo_1[foo_2] = (foo_3 - 1) * (foo_4[foo_5 - 1] + foo_6[foo_7 - 2]);13;
8;exit;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;7;
5;8;
6;5;
7;
